docs/programming-guide.md

```diff
@@ -1353,7 +1353,7 @@ data.map(x -> accum.add(x); f(x););
 <div data-lang="python"  markdown="1">
 {% highlight python %}
 accum = sc.accumulator(0)
-data.map(lambda x => acc.add(x); f(x))
+data.map(lambda x: acc.add(x) and f(x))
```

Comment:
It looks to me like `acc.add(x)` will always evaluate to `None` [1], so the function will never evaluate `f(x)`.  A hacky way to fix this may be to change the lambda to

```python
lambda x: not acc.add(x) and f(x)
```

because `not None` evaluates to `True`, but this certainly won't make things more clear for the reader.  The best fix would be do define a separate function

```python
def g(x):
    acc.add(x)
    return f(x)

data.map(g)
```